index.js

```diff
@@ -709,11 +705,20 @@ class Peer extends stream.Duplex {
     )
     this.emit('iceStateChange', iceConnectionState, iceGatheringState)
 
-    if (iceConnectionState === 'connected' || iceConnectionState === 'completed') {
+    if (iceConnectionState === 'connected' || iceGatheringState === 'completed') {
+      this._isRestarting = false
       this._pcReady = true
       this._maybeReady()
     }
-    if (iceConnectionState === 'failed') {
+
+    if (iceConnectionState === 'failed' && this.iceRestartEnabled) {
+      if (this.initiator && !this._isRestarting) {
+        this._isNegotiating = false
+        this._isRestarting = true
+
+        this._needsNegotiation(true)
```

Comment:
`_needsNegotiation` does not accept a param. The new restart param has only been added to the `negotiate` function. Seems like it needs to also be threaded through the `_needsNegotiation` function right?

Reply:
Think I was trying to pass the restart flag to `negotiate`, not `_needsNegotiation`. Corrected in the latest patch. Now `_needsNegotation` uses the `_isRestarting` flag to seed `negotiate` with the flag.